kategory/src/main/kotlin/kategory/instances/NonEmptyListInstances.kt

```diff
@@ -32,6 +34,23 @@ interface NonEmptyListInstances :
         }
     }
 
+    override fun <A, B> foldL(fa: HK<NonEmptyListHK, A>, b: B, f: (B, A) -> B): B = fa.ev().tail.fold(f(b, fa.ev().head), f)
+
+    override fun <A, B> foldR(fa: HK<NonEmptyListHK, A>, lb: Eval<B>, f: (A, Eval<B>) -> Eval<B>): Eval<B> {
+        fun loop(fa_p: NonEmptyList<A>): Eval<B> = when {
```

Comment:
Is loop stack-safe here? I know in the past we've found differences in implementation from cats in this regard. I see ListKW has the same implementation, maybe it needs a law.